src/transaction/transaction.rs

```diff
@@ -9,9 +9,25 @@ use crate::{
 };
 use derive_new::new;
 use futures::{executor::ThreadPool, prelude::*, stream::BoxStream};
-use std::{iter, mem, ops::RangeBounds, sync::Arc};
+use std::{iter, ops::RangeBounds, sync::Arc};
 use tikv_client_proto::{kvrpcpb, pdpb::Timestamp};
 
+#[derive(PartialEq)]
+enum TransactionStatus {
+    /// The transaction is read-only [`Snapshot`](super::Snapshot::Snapshot), no need to commit or rollback or panic on drop.
+    ReadOnly,
```

Comment:
Do we need to know this explicitly? Could we instead check for an empty buffer?

Reply:
The buffer is protected by an async-aware mutex. So it brings difficulty in implementing `Drop` if we do this.